src/Tests/Microsoft.DotNet.ApiCompatibility.Tests/Rules/EnumValuesMustMatchTests.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.DotNet.ApiCompatibility.Abstractions;
+using Xunit;
+
+namespace Microsoft.DotNet.ApiCompatibility.Tests
+{
+    public class EnumValuesMustMatchTests
+    {
+        [Fact]
+        public static void EnumValuesNotMatchingReported()
+        {
+            string leftSyntax = @"
+namespace CompatTests
+{
+  public enum First {
+    A = 0,
+    B = 1,
+    C = 2,
+    D = 3,
```

Comment:
Are there other scenarios that would make sense to test for, i.e. a different order of enum members?

Reply:
The declared order doesn't matter, except for when they're using automatic numbering. (And then the value would change, which this rule is already looking for)